graphql/src/main/java/com/scalar/db/graphql/server/GraphQlHandler.java

```diff
@@ -0,0 +1,269 @@
+              request.getParameter("variables"));
+      execute(response, input);
+      return;
+    }
+    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+  }
+
+  private Charset getCharset(String charsetName) {
+    if (charsetName == null) {
+      return StandardCharsets.UTF_8;
+    }
+    try {
+      return Charset.forName(charsetName);
+    } catch (IllegalCharsetNameException | UnsupportedCharsetException e) {
+      return StandardCharsets.UTF_8;
+    }
+  }
+
+  private void handleGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    if (request.getParameter("query") != null) {
+      LOGGER.debug("A GET request with a \"query\" parameter received.");
+      ExecutionInput input =
+          createExecutionInput(
+              request.getParameter("query"),
+              request.getParameter("operationName"),
+              request.getParameter("variables"));
+      execute(response, input);
+    } else if (isGraphiqlEnabled) {
+      LOGGER.debug("A GET request without query parameter received. Serving Graphiql HTML.");
+      response.setContentType("text/html");
+      response.setStatus(HttpServletResponse.SC_OK);
+      response.getOutputStream().write(graphiqlHtml);
+    } else {
+      LOGGER.debug("No \"query\" parameter or Graphiql is disabled.");
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    }
+  }
+
+  private ExecutionInput createExecutionInput(Reader reader) {
+    JsonObject jsonObject = JsonParser.parseReader(reader).getAsJsonObject();
+    ExecutionInput.Builder executionInput = ExecutionInput.newExecutionInput();
+    JsonElement query = jsonObject.get("query");
+    if (query != null && query.isJsonPrimitive()) {
+      executionInput.query(query.getAsString());
+    }
+    JsonElement operationName = jsonObject.get("operationName");
+    if (operationName != null && operationName.isJsonPrimitive()) {
+      executionInput.operationName(operationName.getAsString());
+    }
+    JsonElement variables = jsonObject.get("variables");
+    if (variables != null && variables.isJsonObject()) {
+      executionInput.variables(GSON.fromJson(variables, MAP_TYPE));
+    }
+    return executionInput.build();
+  }
+
+  private ExecutionInput createExecutionInput(
+      String query, String operationName, String variables) {
+    ExecutionInput.Builder builder = ExecutionInput.newExecutionInput();
+    if (query != null) {
+      builder.query(query);
+    }
+    if (operationName != null) {
+      builder.operationName(operationName);
+    }
+    Map<String, Object> variablesMap = GSON.fromJson(variables, MAP_TYPE);
+    if (variablesMap != null) {
+      builder.variables(variablesMap);
+    }
+    return builder.build();
+  }
+
+  private void execute(HttpServletResponse response, ExecutionInput executionInput)
+      throws IOException {
+    LOGGER.debug("Executing GraphQL input: " + executionInput);
```

Comment:
We can use a placeholder here as follows:
```suggestion
    LOGGER.debug("Executing GraphQL input: {}", executionInput);
```

Reply:
Thank you for the fix! 🙏